packages/react-button/src/utils/useToggleState.ts

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { useControllableState, useMergedEventCallbacks } from '@fluentui/react-utilities';
+import type { ButtonState } from '../Button';
+import type { ToggleButtonProps, ToggleButtonState } from '../ToggleButton';
+
+export function useToggleState<
+  TToggleButtonProps extends ToggleButtonProps,
+  TButtonState extends Partial<ButtonState>,
+  TToggleButtonState extends ToggleButtonState
+>(props: TToggleButtonProps, state: TButtonState): TToggleButtonState {
+  const { checked, defaultChecked, disabled, disabledFocusable } = props;
+  const { onClick, role } = state.root || {};
+
+  const [checkedValue, setCheckedValue] = useControllableState({
+    state: checked,
+    defaultState: defaultChecked,
+    initialState: false,
+  });
+
+  const isCheckboxTypeRole = role === 'menuitemcheckbox' || role === 'checkbox';
+
+  const onToggleClick = React.useCallback(
+    ev => {
+      if (!disabled && !disabledFocusable) {
+        if (ev.defaultPrevented) {
```

Comment:
What do you think about adding a test case for this?

Reply:
Added!